src/GovUk.Education.ExploreEducationStatistics.Data.Services/ObservationService.cs

```diff
@@ -51,7 +51,7 @@ public class ObservationService : AbstractRepository<Observation, long>, IObserv
             {
                 locationIds = await _context
                     .Location
-                    .Where(LocationPredicateBuilder.Build(query.LocationIds, query.Locations))
+                    .Where(LocationPredicateBuilder.Build(query.LocationIds?.ToList(), query.Locations))
```

Comment:
It seems like we're using `IEnumerable` in most of the `ObservationQueryContext` fields, but it kinda feels like we could replace most of these with `Set` or `List` types (or their interfaces) to avoid needing to fix potential multiple enumerations issues.

I'd probably lean toward `Set` to have built-in dupe removal, but I guess that depends on if order is important (there's always `SortedSet`).

Reply:
I think this is a great idea, and something I've been adding more of in 2999 in fact.  Had a discussion with Mark about it too and the reasoning behind it.  What a shame that IQueryable and IList both extend IEnumerable!

Def happy to extend this pattern beyond this piece of work :)